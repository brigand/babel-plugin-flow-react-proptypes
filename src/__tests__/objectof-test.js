const babel = require('babel-core');
const content = `
var React = require('react');

type Complex = {
  real: number;
  imag: number;
}
type FooProps = {
  oneNumber: number;
  oneComplex: Complex;
  manyNumbers: {[name: string]: number},
  manyComplex: {[name: string]: Complex},
}

export default class Foo extends React.Component {
  props: FooProps
}
`;

it('objectOf', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['env', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
