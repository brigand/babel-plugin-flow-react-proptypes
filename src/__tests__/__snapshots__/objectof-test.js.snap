// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`objectOf 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\"); } return call && (typeof call === \\"object\\" || typeof call === \\"function\\") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== \\"function\\" && superClass !== null) { throw new TypeError(\\"Super expression must either be null or a function, not \\" + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var React = require('react');

var Foo = function (_React$Component) {
  _inherits(Foo, _React$Component);

  function Foo() {
    _classCallCheck(this, Foo);

    return _possibleConstructorReturn(this, (Foo.__proto__ || Object.getPrototypeOf(Foo)).apply(this, arguments));
  }

  return Foo;
}(React.Component);

Foo.propTypes = {
  oneNumber: _propTypes2.default.number.isRequired,
  oneComplex: _propTypes2.default.shape({
    real: _propTypes2.default.number.isRequired,
    imag: _propTypes2.default.number.isRequired
  }).isRequired,
  manyNumbers: _propTypes2.default.objectOf(_propTypes2.default.number.isRequired).isRequired,
  manyComplex: _propTypes2.default.objectOf(_propTypes2.default.shape({
    real: _propTypes2.default.number.isRequired,
    imag: _propTypes2.default.number.isRequired
  }).isRequired).isRequired
};
exports.default = Foo;"
`;
